### ISSUE TITLE
Not compatible with composer 2

### ISSUE THREAD

**Issue by darthsteven** [deviantintegral/composer-gavel#8]
I get a fatal error when trying to write the version constraint into the composer.json:

```
PHP Fatal error:  Uncaught TypeError: Argument 3 passed to Composer\Package\Locker::__construct() must be an instance of Composer\Installer\InstallationManager, instance of Composer\Repository\RepositoryManager given, called in ~/.composer/vendor/deviantintegral/composer-gavel/src/ComposerVersionRequirement.php on line 128 and defined in phar:///usr/local/bin/composer/src/Composer/Package/Locker.php:60
Stack trace:
#0 ~/.composer/vendor/deviantintegral/composer-gavel/src/ComposerVersionRequirement.php(128): Composer\Package\Locker->__construct(Object(Composer\IO\ConsoleIO), Object(Composer\Json\JsonFile), Object(Composer\Repository\RepositoryManager), Object(Composer\Installer\InstallationManager), '{\n\t"name":     ...')
#1 ~/.composer/vendor/deviantintegral/composer-gavel/src/ComposerVersionRequirement.php(90): Deviantintegral\ComposerGavel\ComposerVersionRequirement->writeConstraint('^2.0.8')
#2 [internal function]: Deviantintegral\Com in phar:///usr/local/bin/composer/src/Composer/Package/Locker.php on line 60
```

https://github.com/composer/composer/commit/6059acf0a39fa97147fa8c26de95b4aa8ea1bcc4#diff-fd239736b8febf7e156d99bf35009c8dff1df6c6d2c1a0b65054c5860cf5e40cL53-R51 removed a parameter from the constructor of the Locker.

Which we use here:

https://github.com/deviantintegral/composer-gavel/blob/92fc13163a709408ddab945f5debb91abbc5d365/src/ComposerVersionRequirement.php#L128

So if you try to write the version constraint when using composer 2, you get a PHP fatal error. This is actually on updating the content hash in the `composer.lock` file, so the constraint does actually make it into the `composer.json` file.

Are you wanting to maintain compatibility with composer 1 and composer 2 with the same version of gavel? If so we'll need to do version detection or similar to fix.

**Comment by darthsteven** [comment]
Apparently we can use `PluginInterface::PLUGIN_API_VERSION` to version detect.

**Comment by deviantintegral** [comment]
Thanks for digging into this! I have a PR up at #9 .

**Comment by deviantintegral** [comment]
I've merged the above PR along with #10 and just tagged a `0.4.0` release.
